### PR TITLE
(Workshop Bounty) Logging In With Flask

### PR DESCRIPTION
**Pull request by sohamb117**
- [x]  Workshop that meets all of the "Good workshops" guidelines outlined above (View below)

- [x]  1-2 minute video clip from your test club meeting showing what members built (View: https://hackclub.slack.com/files/U013GCKMN56/F01FAP25VC3/meeting_recording.mp4)

- [x]  All assets (images, PDFs, and any other binaries) must be hosted using the #cdn channel on the Hack Club Slack for posterity and to cut down on repository size.

- [x]  All workshops must be in English at native-level proficiency. Phrasing and grammar must be perfect.

- [x]  You must be either a registered Hack Club leader or an active member of the Slack.

Good workshops guidelines:
Help club members produce a compelling and real project in under 20 minutes

> I tested it with some friends, and it took us roughly 20 minutes to finish.

Are very easy to hack on for long periods of time after completion

> The project not only teaches you how to make an entire backend, but it guides you on where to take this backend, provides demonstrations on how it can be improved, discusses its flaws and benefits, and includes a demonstration with a frontend to show how you can integrate with a frontend. 

Have instructive and compelling images, GIFs, and visuals

> I included lots of images, as well as a demo GIF to show what this does.

Use repl.it or a similar service that allows members to begin coding with one click

> We used repl.it for the whole project, and we used Insomnia which is available as a browser extension to demo and test the project.

Link unknown concepts to external sources.

> There's a list of links at the bottom to documentation and further reading, and links in text throughout to help provide more resources.

End with at least 3 demos of how the project can be expanded, with live links and code links for each demo.

> I included 3 demonstrations with short descriptions. These are done on repl.it with links to allow viewers to run and read my code.

This workshop is aimed at people interested in creating a login system from scratch. It discusses how they work, as well as walking the reader through the creation of a webserver in Flask step by step. It also provides ideas for possible improvements as well as how to integrate it with a frontend, and it's aimed to be a full project that also serves as a starting point for people to build off of and modify into something that's even more useful for their specific case.

### CODE REVIEW

File: workshops/flask-login/README.md [hunk 1]
```diff
@@ -0,0 +1,306 @@
...
+We'll be using [Repl.it](https://repl.it) to work on this, because it makes it very easy for us to run our webserver and code in Python. Let's set up our accounts.
+
+Go to [repl.it](https://repl.it). You should see something like this:
+![repl.it](https://cloud-i5iknsq3z.vercel.app/0image.png)
+
+Either press "Log In" or "Sign Up," depending on whether or not you already have an account. 
+Once you're done, you should see the "Home" page.
+Press either the plus icon in the top right or the "New Repl" button in the top left. You should see this screen: 
+
+![repl create](https://cloud-f0le2m5gi.vercel.app/0image.png)
```
Comment (sampoder) [inline workshops/flask-login/README.md:48]:
```suggestion
![repl create](https://cloud-f0le2m5gi.vercel.app/0image.png)

```

**Review by sampoder — CHANGES_REQUESTED** [review]
Hey @sohamb117!

Thanks for submitting this workshop, I think this is certainly a cool workshop but I've got one main concern before thinking we can merge this. 

I think it'd be really cool if someone could actually apply what they learn in the workshop in a real life project super quickly.

So I was thinking, if you could make the following three changes then we'd actually have a semi-secure app:

- Switch from JSON to repl.it database, ask @KhushrajRathod (he knows a bit about it)
- Have some system for encrypting the passwords
- Use [Flask Login](https://flask-login.readthedocs.io/en/latest/) to make a sample protected route

If you were able to do that, then I think this would be an epic workshop that'd students could take a lot from and apply in the real world.

As we normally do with workshops, I'm going to close this but feel free to reopen when you think you've got something

Thank you,

Sam 🚀

**Comment by sampoder** [comment]
Hiiii sorry I'm clumsy, I meant to close this :joy:

**Comment by KhushrajRathod** [comment]
The amount of confusion in my head at this point was unimaginable
Screenshot 2020-12-02 at 5 59 05 PM

**Pull request merged**